scalafix-core/shared/src/main/scala/scalafix/internal/util/LazySymbolTable.scala

```diff
@@ -0,0 +1,117 @@
+package scalafix.internal.util
+
+import java.io.IOException
+import scala.meta.internal.semanticdb3.Scala.Symbols
+import java.net.URI
+import java.nio.file.FileSystem
+import java.nio.file.FileSystemAlreadyExistsException
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import org.langmeta.internal.io.PathIO
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.Classpath
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.semanticdb3.Scala._
+import scala.meta.internal.{semanticdb3 => s}
+import scala.util.control.NonFatal
+
+/**
+  * Implementation of SymbolTable that lazily loads symbols on demand.
+  *
+  * @param mclasspath The classpath to load symbols from. Needs to be pre-processed by metacp,
+  *                   see ClasspathOps.toMclasspath in scalafix-cli.
+  */
+class LazySymbolTable(mclasspath: Classpath) extends SymbolTable {
+
+  // Map from symbols to the paths to their corresponding semanticdb file where they are stored.
+  private val unloadedSymbols = TrieMap.empty[String, AbsolutePath]
```

Comment:
"unloaded" as in "not yet loaded" or as in "previously loaded, but no longer available"?

Reply:
Renamed to `notYetLoadedSymbols`